targets/unity-v2/templates/InstanceAPI.cs.ejs

```diff
@@ -0,0 +1,91 @@
+#if <%- getApiDefineFlag(api) %>
+using System;
+using System.Collections.Generic;
+using PlayFab.<%- api.name %>Models;
+using PlayFab.Internal;
+using PlayFab.Json;
+using PlayFab.Public;
+
+namespace PlayFab
+{
+<%- generateApiSummary("    ", api, "description")
+%>    public <%- isPartial %>class PlayFab<%- api.name %>InstanceAPI
+    {
+        private PlayFabApiSettings apiSettings = null;
+        private PlayFabAuthenticationContext authenticationContext = null;
+
+        public PlayFab<%- api.name %>InstanceAPI() {}
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabApiSettings settings) 
+        {
+            apiSettings = settings;
+        }
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabAuthenticationContext context) 
+        {
+            authenticationContext = context;
+        }
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabApiSettings settings, PlayFabAuthenticationContext context) 
+        {
+            apiSettings = settings;
+            authenticationContext = context;
+        }
+
+        public void SetSettings(PlayFabApiSettings settings)
+        {
+            apiSettings = settings;
+        }
+
+        public PlayFabApiSettings GetSettings()
+        {
+            return apiSettings;
+        }
+
+        public void SetAuthenticationContext(PlayFabAuthenticationContext context)
+        {
+            authenticationContext = context;
+        }
+
+        public PlayFabAuthenticationContext GetAuthenticationContext()
+        {
+            return authenticationContext;
+        }
+
+<% if (hasClientOptions) { %>
+        /// <summary>
+        /// Check to See if the client is logged in.
+        /// </summary>
+        public bool IsClientLoggedIn()
+        {
+            return PlayFabHttp.IsClientLoggedIn();
+        }
+<% }%>
+<% if (hasEntityTokenOptions) { %>
+        /// <summary>
+        /// Check to See if the entity is logged in.
+        /// </summary>
+        public bool IsEntityLoggedIn()
+        {
+            var transport = PluginManager.GetPlugin<IPlayFabTransportPlugin>(PluginContract.PlayFab_Transport);
+            return !string.IsNullOrEmpty(transport.EntityToken);
```

Comment:
Really good catch, Paul. Thank you! I checked other SDKs and only C# has a problem (a bit different one: this method is checking PlayFabSettings there instead of checking authentication context). C++ and Unreal are fine, they have correct code.

---
In reply to: [261837473](https://github.com/PlayFab/SDKGenerator/pull/420#discussion_r261837473) [](ancestors = 261837473)

Reply:
Sure I will